map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java

```diff
@@ -29,116 +36,142 @@
 
-      worldTimeZonesLabel.getStyleClass().add("panel-label");
-      censusLabel.getStyleClass().add("panel-label");
-      facilitiesLabel.getStyleClass().add("panel-label");
+      // create a label to display the view status
+      Label layerViewStatusLabel = new Label("Current view status: ");
 
-      // add labels to the control panel
-      controlsVBox.getChildren().addAll(worldTimeZonesLabel, censusLabel, facilitiesLabel);
+      // create a checkbox UI that toggles the visibility of the feature layer
+      CheckBox visibilityCheckBox = new CheckBox();
+      visibilityCheckBox.setText("Layer active");
+      visibilityCheckBox.setSelected(true);
+      visibilityCheckBox.setDisable(true);
+      visibilityCheckBox.setOnAction(event -> featureLayer.setVisible(visibilityCheckBox.isSelected()));
 
-      // create three layers to add to the ArcGISMap
-      final ArcGISTiledLayer tiledLayer = new ArcGISTiledLayer(SERVICE_TIME_ZONES);
+      // create a button and a listener to load a feature layer
+      Button loadLayerButton = new Button("Load Layer");
 
-      // this layer will be OUT_OF_SCALE outside of its set min/max scales
-      final ArcGISMapImageLayer imageLayer = new ArcGISMapImageLayer(SERVICE_CENSUS);
-      imageLayer.setMinScale(MIN_SCALE);
-      imageLayer.setMaxScale(MIN_SCALE / 10);
+      loadLayerButton.setOnAction(event -> {
 
-      // creating a layer from a service feature table
-      final ServiceFeatureTable featureTable = new ServiceFeatureTable(SERVICE_RECREATION);
-      final FeatureLayer featureLayer = new FeatureLayer(featureTable);
+        if (!visibilityCheckBox.isDisabled()) {
+          visibilityCheckBox.setDisable(true);
+        }
 
-      // adding layers to the ArcGISMaps' layer list
-      final ArcGISMap map = new ArcGISMap();
-      map.getOperationalLayers().add(tiledLayer);
-      map.getOperationalLayers().add(imageLayer);
-      map.getOperationalLayers().add(featureLayer);
+        if (featureLayer != null) {
+          map.getOperationalLayers().clear();
+        }
 
-      // create a view and set ArcGISMap to it
-      mapView = new MapView();
-      mapView.setMap(map);
+        // create a new feature layer from a portal item
+        final PortalItem portalItem = new PortalItem(new Portal("https://runtime.maps.arcgis.com/"),
+          "b8f4033069f141729ffb298b7418b653");
+        featureLayer = new FeatureLayer(portalItem, 0);
+        //  set a minimum and a maximum scale for the visibility of the feature layer
+        featureLayer.setMinScale(400000000);
+        featureLayer.setMaxScale(4000000);
+        // add the feature layer to the map
+        map.getOperationalLayers().add(featureLayer);
+
+        // add a done loading listener to the feature layer
+        featureLayer.addDoneLoadingListener(() -> {
+          if (featureLayer.getLoadStatus() == LoadStatus.LOADED){
```

Comment:
extra space needed here between the parenthesis and curly brackets 
```suggestion
          if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
```